.github/workflows/update_apps.yml

```diff
@@ -0,0 +1,58 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Update_Apps
+
+# Controls when the workflow will run
+on:  
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  update-apps:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Runs a set of commands using the runners shell
+      - name: Run a multi-line script
+        run: |
+          #source pi-apps functions
+          DIRECTORY=$GITHUB_WORKSPACE
+          set -a
+          source $GITHUB_WORKSPACE/api
+
+          #add special functions
+          get_release() {
+            curl --silent "https://api.github.com/repos/$1/releases/latest" | jq -r '.tag_name' | sed s/v//g
+          }
+
+          set +a
+          #make sure all update scripts are executable
+          chmod +x $GITHUB_WORKSPACE/.github/workflows/updates/*.sh
+          cd $GITHUB_WORKSPACE
+          apps=( .github/workflows/updates/*.sh )
+          for app_directory in ${apps[@]}; do
+            app_name=$(echo ${app_directory%.*} | sed 's:.*/::')
+            status "Checking $app_name for updates"
+            # move to app folder
+            cd "./apps/$app_name"
+            # run app update script
+            "$GITHUB_WORKSPACE/$app_directory"
+          done
+          cd
+
+      - name: Push changes to repo
+        uses: EndBug/add-and-commit@v7 # You can change this to use a specific version.
+        with:
+
+          default_author: github_actions
+
+          # The message for the commit.
+          # Default: 'Commit from GitHub Actions (name of the workflow)'
+          message: 'Automatic update'
```

Comment:
Maybe have something a little more descriptive in the message, Like `Automatic update, run by GitHub Actions`.

It'd also be cood if it generatd a list of all apps updated by the workflow!

Reply:
yeah I'm still working on this
adding a check to verify that the new version actually exists for the armhf/arm64 (I've seen a couple of github releases drop support for armhf due to issues and don't want to update the version variable if the file doesn't actually exist)